core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -244,5 +244,13 @@ private void resolvePipelineAggregatorOrder(Map<String, AggregatorFactory> aggFa
                 orderedPipelineAggregators.add(factory);
             }
         }
+
+        AggregatorFactory[] getAggregatorFactories() {
+            return this.factories.toArray(new AggregatorFactory[this.factories.size()]);
+        }
+
+        List<PipelineAggregatorFactory> getPipelineAggregatorFactories() {
```

Comment:
agreed, should I return a list from both and do the conversion to an array in the AggParsers class?